File: Offer/2522/main.py
import collections
from typing import List


class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[int]:
        if root == None:
            return []

        array: List[int] = []
        queue = collections.deque([root])
        node: TreeNode
        while len(queue) != 0:
            node = queue.popleft()
            array.append(node.val)
            if node.left != None:
                queue.append(node.left)
            if node.right != None:
                queue.append(node.right)
        return array

File: Offer/2522/main.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func levelOrder(root *TreeNode) []int {
	array := []int{}
	if root == nil {
		return array
	}

	queue := []*TreeNode{root}
	var node *TreeNode
	for len(queue) != 0 {
		node = queue[0]
		queue = queue[1:]

		array = append(array, node.Val)

		if node.Left != nil {
			queue = append(queue, node.Left)
		}
		if node.Right != nil {
			queue = append(queue, node.Right)
		}
	}
	return array
}
